flatland/envs/rewards.py

```diff
@@ -188,16 +189,67 @@ def step_reward(self, agent: EnvAgent, agent_transition_data: AgentTransitionDat
         if agent.handle in self._previous_speeds:
             smoothness = -(current_speed - self._previous_speeds[agent.handle]) ** 2
         self._previous_speeds[agent.handle] = current_speed
-        return [default_reward, energy_efficiency, smoothness]
+        return default_reward, energy_efficiency, smoothness
 
-    def end_of_episode_reward(self, agent: EnvAgent, distance_map: DistanceMap, elapsed_steps: int) -> List[float]:
+    def end_of_episode_reward(self, agent: EnvAgent, distance_map: DistanceMap, elapsed_steps: int) -> Tuple[float, float, float]:
         default_reward = super().end_of_episode_reward(agent=agent, distance_map=distance_map, elapsed_steps=elapsed_steps)
         energy_efficency = 0
         smoothness = 0
-        return [default_reward, energy_efficency, smoothness]
+        return default_reward, energy_efficency, smoothness
 
-    def cumulate(self, *rewards: List[float]) -> List[float]:
-        return [sum([r[i] for r in rewards]) for i in range(3)]
+    def cumulate(self, *rewards: Tuple[float, float, float]) -> Tuple[float, float, float]:
+        return tuple([sum([r[i] for r in rewards]) for i in range(3)])
 
-    def empty(self) -> List[float]:
-        return [0] * 3
+    def empty(self) -> Tuple[float, float, float]:
+        return 0, 0, 0
+
+
+class PunctualityRewards(Rewards[Tuple[int, int]]):
+    """
+    Punctuality: n_stops_on_time / n_stops
```

Comment:
```suggestion
    Punctuality: n_stops_on_time / n_stops
    ```suggestion
    Punctuality: n_stops_on_time / n_stops
    An agent is deemed not punctual at a stop if it arrives to late, departs to early or does not serve the stop at all. If an agent is punctual at a stop, n_stops_on_time is increased by 1.
```
```